### DISCUSSION TITLE
Err.Raise for all the handled errors

### DISCUSSION THREAD

**Discussion by 6DiegoDiego9** [GCuser99/SeleniumVBA#20] [General]
I saw that you used many times the very basic Msgbox+End combo for your handled errors.
For example:
```
MsgBox "The following specified reference path does not exist: """ & refPath & """" 
End
```

I'd suggest to use the proper VBA instruction:
```
Err.Raise 76, , "The following specified reference path does not exist: """ & refPath & """"
```

Advantages:
- the users can choose whether to "End" or "Debug"
- projects using global error handlers (eg. I use the fabulous vbWatchDog) can channel the error to it so that it's displayed and handled as globally defined. For example, in my case, a multi-colored message box with a predefined header and useful info like the module name and procedure name and line number and content that generated it and the entire call stack.

Note: 76 is just the native error number that I found in the VBA Error Codes list for "path not found" errors but it's not important. You can use any number, even always 1 for faster (lazy) coding, because as long as you specify a Description it overrides the native description.
My personal preference is to generally always use the number 1 for all the errors internal to the code (bug in code) and 2 for all the errors external to the code (user's fault), usually with explicit custom descriptions.  Depending on this number, my global error handler uses a different graphical and textual template (eg header) for displaying the error.

**Comment by GCuser99** [comment]
@6DiegoDiego9, if you're interested in designing/writing the error handling functionality for this code (or any other accepted coding style) then you are very welcome to. For me this is a collaboration (of which I borrowed much from previous collaborations) and after I publish the Add-in version, I will probably be approaching a point of diminishing contribution given my background, skills, and interest. So go at it!
